storage/core/src/main/java/com/hortonworks/streamline/storage/filestorage/DbFileStorage.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.storage.filestorage;
+
+import com.hortonworks.streamline.common.util.FileStorage;
+import com.hortonworks.streamline.storage.StorageManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * DB based file storage that stores the file as a blob.
+ */
+public class DbFileStorage implements FileStorage {
+    private static final Logger LOG = LoggerFactory.getLogger(DbFileStorage.class);
+    private StorageManager dao;
+
+    @Override
+    public void init(Map<String, String> config) {
+    }
+
+    @Override
+    public String uploadFile(InputStream inputStream, String name) throws IOException {
+        LOG.debug("Upload '{}'", name);
+        Optional<FileBlob> existing = get(name);
+        if (existing.isPresent()) {
+            LOG.debug("Trying to update existing file '{}'", name);
+            FileBlob updated = new FileBlob(existing.get());
+            updated.setData(inputStream);
```

Comment:
I had initially wrapped the inputStream in `BufferedInputStream` but later decided to remove it and leave it to the caller. I missed removing it from the add part. Will remove it.